backend/accounts.go

```diff
@@ -257,12 +257,31 @@ func (backend *Backend) nextAccountNumber(coinCode coinpkg.Code, keystore keysto
 		return account.Configurations.ContainsRootFingerprint(rootFingerprint)
 	}
 
-	nextAccountNumber := uint16(len(backend.filterAccounts(accountsConfig, filter)))
+	nextAccountNumber := uint32(0)
+	for _, account := range backend.filterAccounts(accountsConfig, filter) {
+		if len(account.Configurations) == 0 {
+			backend.log.WithField("code", account.Code).Error("expected at least one signing configuration")
+			continue
+		}
+		accountNumber, err := account.Configurations[0].AccountNumber()
+		if err != nil {
+			backend.log.WithField("code", account.Code).Error("could not parse account number")
+			continue
+		}
+		if accountNumber > nextAccountNumber {
+			nextAccountNumber = accountNumber
+		}
+	}
+	nextAccountNumber++
+
+	if !keystore.SupportsMultipleAccounts() && nextAccountNumber >= 1 {
+		return 0, errp.WithStack(ErrAccountLimitReached)
+	}
 
 	if nextAccountNumber >= accountsHardLimit {
 		return 0, errp.WithStack(ErrAccountLimitReached)
 	}
-	return nextAccountNumber, nil
+	return uint16(nextAccountNumber), nil
```

Comment:
I know this cannot happen (famous last words), but this is a theoretical unchecked int overflow.